src/test/java/org/apache/commons/pool2/impl/AtomicIntegerFactory.java

```diff
@@ -0,0 +1,117 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.commons.pool2.impl;
+
+import java.util.concurrent.atomic.AtomicInteger;
+
+import org.apache.commons.pool2.BasePooledObjectFactory;
+import org.apache.commons.pool2.PooledObject;
+import org.apache.commons.pool2.impl.DefaultPooledObject;
+
+class AtomicIntegerFactory
```

Comment:
This class could use a Javadoc to explain how it fits in.

Reply:
It is just a test factory.  It was pulled out of TestGenericObjectPool, which still uses it.  I pulled it out so TestBaseGenericObjectPool could use it directly.  As a side note, since the refactoring to move some things to BGOP, it would be good to migrate associated tests to TBGOP.

Reply:
Just added class javadoc.